### PR TITLE
[universe]: fix proof sync for large asset minting batches

### PR DESCRIPTION
**Pull request by guggero**
I encountered this on `testnet` while trying to reproduce https://github.com/lightninglabs/taproot-assets/issues/675.

Can be reproduced with:
```
--universe.sync-all-assets
--universe.federationserver=testnet.universe.lightning.finance:10029
```

And `tapcli u s --universe_host testnet.universe.lightning.finance:10029`.

This PR fixes the issue by making sure any group anchors are inserted into the DB first.

### CODE REVIEW

File: universe/syncer.go [hunk 1]
```diff
@@ -319,6 +308,28 @@ func (s *SimpleSyncer) syncRoot(ctx context.Context, remoteRoot Root,
...
+					)
+					if err != nil {
+						return fmt.Errorf("unable to "+
+							"register group "+
+							"anchor proof: %w", err)
+					}
+
+					// No need to batch this further, we've
+					// already inserted it.
+					return nil
```
Comment (jharveyb) [inline universe/syncer.go:330]: Suggested fix:

```go
newAnchorProofs  = append(
     newAnchorProofs, leafProof.Leaf,
)
```

File: universe/syncer.go [hunk 2]
```diff
@@ -338,6 +349,17 @@ func (s *SimpleSyncer) syncRoot(ctx context.Context, remoteRoot Root,
 		return err
 	}
 
+	// We use an error group to simply the error handling of a goroutine.
+	// This goroutine will handle reading in batches of new leaves to
+	// insert into the DB. We'll fee the output of the goroutines below
+	// into the input fetchedLeaves channel.
+	batchSyncEG.Go(func() error {
+		newLeafProofs, err = s.batchStreamNewItems(
```
Comment (jharveyb) [inline universe/syncer.go:357]: Itest failure is caused by the leaves being inserted above not being added to `newLeafProofs`, which is used to build the RPC response.

I verified a fix locally by declaring `newAnchorProofs`, populating that as suggested above, and then joining the two lists before sending the `AssetSyncDiff`:

```go
newLeafProofs = append(newAnchorProofs, newLeafProofs...)
```

**Review by jharveyb — COMMENTED** [review]
Looks good! Just need that one fix + possible logging improvements.

File: universe/syncer.go (hunk 1 above)
Comment (guggero) [reply universe/syncer.go:330]: Thanks, great catch!

**Review by jharveyb — APPROVED** [review]
LGTM 👍🏽

**Review by Roasbeef — APPROVED** [review]
LGTM 🚜

**Pull request merged**